In [1]:
from google.colab import files
uploaded = files.upload()

Saving credit_cleaned.csv to credit_cleaned.csv


In [3]:
import pandas as pd

df = pd.read_csv('credit_cleaned.csv')

# Binary encoding
df['person_gender'] = df['person_gender'].map({'female': 0, 'male': 1})
df['previous_loan_defaults_on_file'] = df['previous_loan_defaults_on_file'].map({'No': 0, 'Yes': 1})

# Ordinal encoding for education (natural order)
education_order = {'High School': 0, 'Associate': 1, 'Bachelor': 2, 'Master': 3, 'Doctorate': 4}
df['person_education'] = df['person_education'].map(education_order)

# One-hot encoding for home_ownership and loan_intent (no natural order)
df = pd.get_dummies(df, columns=['person_home_ownership', 'loan_intent'], prefix=['home', 'intent'])

# Convert one-hot boolean columns to 0/1
bool_cols = df.select_dtypes(include='bool').columns
df[bool_cols] = df[bool_cols].astype(int)

df.to_csv('credit_encoded.csv', index=False)
print(df.shape)
df.head()

(44990, 22)


,person_age,person_gender,person_education,person_income,person_emp_exp,loan_amnt,loan_int_rate,loan_percent_income,cb_person_cred_hist_length,credit_score,...,home_MORTGAGE,home_OTHER,home_OWN,home_RENT,intent_DEBTCONSOLIDATION,intent_EDUCATION,intent_HOMEIMPROVEMENT,intent_MEDICAL,intent_PERSONAL,intent_VENTURE
0,22.0,0,3,71948.0,0,35000.0,16.02,0.49,3.0,561,...,0,0,0,1,0,0,0,0,1,0
1,21.0,0,0,12282.0,0,1000.0,11.14,0.08,2.0,504,...,0,0,1,0,0,1,0,0,0,0
2,25.0,0,0,12438.0,3,5500.0,12.87,0.44,3.0,635,...,1,0,0,0,0,0,0,1,0,0
3,23.0,0,2,79753.0,0,35000.0,15.23,0.44,2.0,675,...,0,0,0,1,0,0,0,1,0,0
4,24.0,1,3,66135.0,1,35000.0,14.27,0.53,4.0,586,...,0,0,0,1,0,0,0,1,0,0


**Encoding Notes**

- Started with the cleaned dataset: 44,990 rows, 14 columns.
- Gender and previous defaults: turned into 0s and 1s, since each only has two options.
- Education: turned into numbers 0 through 4, in order from High School to Doctorate, so the ranking is preserved.
- Home ownership and loan intent: split into separate yes/no columns for each category, since there's no natural order between them.
- Loan status (our target) was left as is.
- Final dataset: 44,990 rows, 22 columns, the extra columns coming from splitting home ownership and loan intent.